lib/jsonapi_compliable/adapters/abstract.rb

```diff
@@ -76,17 +76,180 @@ module Adapters
     # @see Adapters::ActiveRecordSideloading
     # @see Adapters::Null
     class Abstract
-      # @param scope The scope object we are chaining
-      # @param [Symbol] attribute The attribute name we are filtering
-      # @param value The corresponding query parameter value
-      # @return the scope
-      #
-      # @example ActiveRecord default
-      #   def filter(scope, attribute, value)
-      #     scope.where(attribute => value)
-      #   end
-      def filter(scope, attribute, value)
-        raise 'you must override #filter in an adapter subclass'
+      def filter_string_eq(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_eq)
+      end
+
+      def filter_string_eql(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_eql)
+      end
+
+      def filter_string_not_eq(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_not_eq)
+      end
+
+      def filter_string_not_eql(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_not_eql)
+      end
+
+      def filter_string_prefix(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_prefix)
+      end
+
+      def filter_string_not_prefix(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_not_prefix)
+      end
+
+      def filter_string_suffix(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_suffix)
+      end
+
+      def filter_string_not_suffix(scope, attribute, value)
+        raise Errors::AdapterNotImplemented.new(self, attribute, :filter_string_not_suffix)
+      end
+
+      def filter_string_like(scope, attribute, value)
```

Comment:
“Like” is a bit too database-specific for my liking. Taking a cue from the popular Ransack gem, how about “cont” instead?  https://github.com/activerecord-hackery/ransack/blob/master/README.md#search-matchers

Reply:
`cont` is pretty non-obvious to me. How about [matches](https://www.rubydoc.info/github/rails/arel/master/Arel/Predications), taken from arel?

Also this is separate but wanted to mention: I also think other adapters (like elastic) will implement stuff like `fuzzy`, which I kept out so you'd have predictable behavior even moving between datastores.